File: superstitioApi/java/superstitioapi/SuperstitioApiSetup.java
package superstitioapi;

import basemod.BaseMod;
import basemod.interfaces.EditStringsSubscriber;
import com.evacipated.cardcrawl.modthespire.lib.SpireInitializer;
import com.megacrit.cardcrawl.core.Settings;
import com.megacrit.cardcrawl.localization.PowerStrings;

@SpireInitializer
public class SuperstitioApiSetup implements EditStringsSubscriber {

    public static final String MOD_NAME = "SuperstitioApi";

    public SuperstitioApiSetup() {
        // 这里注册颜色
        BaseMod.subscribe(this);
    }

    public static void initialize() {
        new SuperstitioApiSetup();
    }


    @Override
    public void receiveEditStrings() {
        BaseMod.loadCustomStringsFile(PowerStrings.class, DataUtility.makeLocalizationPath(Settings.language, "power"));
    }
}

File: superstitioApi/java/superstitioapi/SuperstitioApiSetup.kt
package superstitioapi

import basemod.BaseMod
import basemod.interfaces.EditStringsSubscriber
import com.evacipated.cardcrawl.modthespire.lib.SpireEnum
import com.evacipated.cardcrawl.modthespire.lib.SpireInitializer
import com.megacrit.cardcrawl.actions.AbstractGameAction.AttackEffect
import com.megacrit.cardcrawl.core.Settings
import com.megacrit.cardcrawl.localization.PowerStrings

@SpireInitializer
class SuperstitioApiSetup : EditStringsSubscriber
{
    init
    {
        BaseMod.subscribe(this)
        Logger.run("Done $this subscribing")
    }

    override fun receiveEditStrings()
    {
        BaseMod.loadCustomStringsFile(
            PowerStrings::class.java,
            DataUtility.makeLocalizationPath(Settings.language, "power")
        )
    }

    object DamageEffect
    {
        @SpireEnum
        lateinit var HeartMultiInOne: AttackEffect
    }

    companion object
    {
        const val MOD_NAME: String = "SuperstitioApi"

        @JvmStatic
        fun initialize()
        {
            val mod = SuperstitioApiSetup()
        }
    }
}
